src/libraries/Common/src/System/Security/Cryptography/DSASecurityTransforms.cs

```diff
@@ -261,9 +261,16 @@ private static SafeSecKeyRefHandle ImportKey(DSAParameters parameters)
 
                 public override byte[] CreateSignature(byte[] rgbHash)
                 {
+                    const int Sha1DigestSizeBytes = 20;
+
                     if (rgbHash == null)
                         throw new ArgumentNullException(nameof(rgbHash));
 
+                    // MacOS only supports FIPS-186-2, which only defines SHA1
+                    // as a supported algorithm.
+                    if (rgbHash.Length != Sha1DigestSizeBytes)
```

Comment:
This is the line that is the breaking change. If we want to preserve the existing behavior, this should then be:
```suggestion
                    if (rgbHash.Length > Sha1DigestSizeBytes)
```